src/rest.cpp

```diff
@@ -590,6 +590,51 @@ static bool rest_chaininfo(const std::any& context, HTTPRequest* req, const std:
     }
 }
 
+RPCHelpMan getdeploymentinfo();
+
+static bool rest_deploymentinfo(const std::any& context, HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req)) return false;
+
+    std::string hashStr;
+    const RESTResponseFormat rf = ParseDataFormat(hashStr, strURIPart);
+
+    switch (rf) {
+    case RESTResponseFormat::JSON: {
+        JSONRPCRequest jsonRequest;
+        jsonRequest.context = context;
+        jsonRequest.params = UniValue(UniValue::VARR);
+
+        if (!hashStr.empty()) {
+            uint256 hash;
+            if (!ParseHashStr(hashStr, hash)) {
+                return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + hashStr);
+            }
+
+            const uint256 hash_bi{ParseHashV(hashStr, "blockhash")};
+            const ChainstateManager* chainman = GetChainman(context, req);
+            if (!chainman) return false;
+            if (!WITH_LOCK(::cs_main, return chainman->m_blockman.LookupBlockIndex(hash_bi))) {
+                return RESTERR(req, HTTP_BAD_REQUEST, "Block not found");
+            }
+
+
+            jsonRequest.params.pushKV("blockhash", hashStr);
+        }
+
+        const UniValue deployment_info_object = getdeploymentinfo().HandleRequest(jsonRequest);
+        const std::string str_json = deployment_info_object.write() + "\n";
+        req->WriteHeader("Content-Type", "application/json");
+        req->WriteReply(HTTP_OK, str_json);
+        return true;
```

Comment:
Looks like this section could be simplified further by dropping two temporaries that are each used only once and that don't seem to make the code clearer.
```diff
-        const UniValue deployment_info_object = getdeploymentinfo().HandleRequest(jsonRequest);
-        const std::string str_json = deployment_info_object.write() + "\n";
         req->WriteHeader("Content-Type", "application/json");
-        req->WriteReply(HTTP_OK, str_json);
+        req->WriteReply(HTTP_OK, getdeploymentinfo().HandleRequest(jsonRequest).write() + "\n");
         return true;
```

Reply:
Nice! Done!